Fix Numerous Spelling, Style, English Issues

waprin opened the pull request:
A 30 min quick, but slow and deliberate, review of the existing codebase.

- Multiple typos and misspellings 
- Multiple missing or incorrect punctuation
- Unused variables

The library is also wildly inconsistent  on its usage of % vs the format function for  string interpolation. I suggest format, but that belongs in its own PR.

http://stackoverflow.com/questions/12382719/python-way-of-printing-with-format-or-percent-form

waprin commented:
@dhermes @tseaver

dhermes commented:
Cool to accept typo / grammar fixes but I'm not sure when punctuation is warranted / if we want to have an opinion on that. Similar on `%` vs. `format`. I used to be all `format` all the time but found it's hard to adhere to in practice (e.g. `logging.log()` assumes `%` formatting)

jonparrott commented:
> hard to adhere to in practice  (e.g. logging.log() assumes % formatting)

I just ignore logging's sugar. `.format` is definitely preferable.

This change LGTM.

dhermes commented on gcloud/dns/changes.py:
```diff
@@ -221,7 +221,8 @@ def create(self, client=None):
     def exists(self, client=None):
-        """API call:  test for the existence of the change set via a GET request
+        """API call:  test for the existence of the change set via a GET
+        request.
```
This is bad. Single-line descriptions are supposed to fit on a single line. So either change the wording or leave it as-is. The 80 char limit is also relaxed on docstrings.

jonparrott replied:
It's up to you, but I really want to call out the fact that the 'single line description should be a single line' is broken *everywhere* in the Python community. Requests, urllib3, github3.py, etc. Sphinx has explicit support for the 'short description' being multiple lines.

dhermes replied:
But why exacerbate that brokenness if we don't have to?

jonparrott replied:
(This particular case might be able to be reduced to one string, so just consider this a meta conversation)

Considering that the community has gone one way, I'd argue that PEP257 is what's broken. Practicality over dogma. The justification for it in PEP257 is

> The summary line may be used by automatic indexing tools; it is important that it fits on one line and is separated from the rest of the docstring by a blank line.

Considering that sphinx has long supported 'breaking' this rule that justification no longer seems totally relevant. I've written documentation for lots of projects and really the *most* important thing is getting your point across clearly, not hard and fast rules. If a slightly longer docstring helps someone understand what a method is *without having to read the entire docstring* then it's win. I definitely agree it should be *one sentence*.

tseaver replied:
@jonparrott ISTM that breaking with the PEP is just wrong: the "single line summary" discipline *works* as a way to focus intent.  When I find myself creeping up to the limit, finding a way to shorten that line (maybe by moving some content down to a more "narrative" paragraph) always improves it.

jonparrott replied:
Sure, as a general guideline it's fine. But it isn't so universally true to
be considered dogma. So many projects have rightfully broke it in order to
have better docs (flask and requests being notable examples).

On Thu, Aug 18, 2016, 5:59 PM Tres Seaver <notifications@github.com> wrote:

> In gcloud/dns/changes.py
> <https://github.com/GoogleCloudPlatform/gcloud-python/pull/2139#discussion_r75414368>
> :
>
> > @@ -221,7 +221,8 @@ def create(self, client=None):
> >          self._set_properties(api_response)
> >
> >      def exists(self, client=None):
> > -        """API call:  test for the existence of the change set via a GET request
> > +        """API call:  test for the existence of the change set via a GET
> > +        request.
>
> @jonparrott <https://github.com/jonparrott> ISTM that breaking with the
> PEP is just wrong: the "single line summary" discipline *works* as a way
> to focus intent. When I find myself creeping up to the limit, finding a way
> to shorten that line (maybe by moving some content down to a more
> "narrative" paragraph) always improves it.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-python/pull/2139/files/180fa5b7f29f5a3815ea33c927c577565035a8e5#r75414368>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAPUc5lClluv85r2hYoETEJS87Xe9Io2ks5qhP_lgaJpZM4Jn88w>
> .
>

dhermes commented on gcloud/pubsub/connection.py:
```diff
@@ -21,7 +21,7 @@
 
 class Connection(base_connection.JSONConnection):
-    """A connection to Google Cloud Pubsub via the JSON REST API.
+    """A connection to Google Cloud Pub/Sub via the JSON REST API.
```
:+1:

dhermes commented on gcloud/storage/batch.py:
```diff
@@ -232,7 +232,7 @@ def _finish_futures(self, responses):
 
     def finish(self):
-        """Submit a single `multipart/mixed` request w/ deferred requests.
+        """Submit a single `multipart/mixed` request with deferred requests.
```
Why was this change necessary?

waprin replied:
Better to avoid abbreviations of common words with international audiences.

dhermes commented on gcloud/resource_manager/client.py:
```diff
@@ -49,7 +49,7 @@ class Client(BaseClient):
 
     def new_project(self, project_id, name=None, labels=None):
-        """Creates a :class:`.Project` bound to the current client.
+        """Create a :class:`.Project` bound to the current client.
```
Why was this necessary?

waprin replied:
Repo owners need to decide whether english language usage is something you care about or not and consistently apply that decision. If you do, then consistent verb tense should be fair game.

dhermes replied:
> english language usage

Out of my depth there. Seems a real challenge to get lots of contributors on the same page with that (without someone dedicated to docs).

dhermes commented:
OK finished my pass and don't see many problems. I'd really like to hear the why on adding all the `.` but the typos are obvious so thanks.

dhermes commented:
@waprin I see you made a new commit since I originally reviewed. Care to point me to what (in the 28 files) has changed?

waprin commented:
My bad I forgot to push commits again. I fixed the line that I moved to the next lien by removing an extraneous space. Will  remember to push commits. 

I don't know about the punctuation, depends on whether docstring consistency is important. Most of the repo uses periods at the end of each docstring param description.

dhermes commented:
We're good to merge here / LGTM

I'm just trying to point out that human language is hard and enforcing these sorts of rules without a tool of some kind seems to be a big stretch goal.

waprin commented:
Yes I agree 1000% that the more we can do with tools the better.
